mlflow2seldon/api.py

```diff
@@ -0,0 +1,291 @@
+
+    seldon_models: Dict[str, _DeployedModel] = dict()
+    try:
+        while True:
+            mlflow_models: Dict[str, _DeployedModel] = dict()
+            logging.info(f"Polling {mlflow_host}")
+            try:
+                # iterate over all registered models ("Models" tab in MLflow WebUI)
+                for model in mlflow_client.search_registered_models():
+                    deploy_image = model.tags.get(deploy_image_tag)
+                    if deploy_image:
+                        deploy_image_ref = neuro_client.parse.remote_image(
+                            deploy_image
+                        ).as_docker_url()
+                    else:
+                        deploy_image_ref = default_image_ref
+                    for model_version in model.latest_versions:
+                        if model_version.current_stage not in ("Staging", "Production"):
+                            # we deploy only Staging and Production models
+                            continue
+                        # Assumption: artifact store in MLflow is platform storage
+                        # mlflow-config related path, e.g.
+                        # ('/', 'usr', 'local', 'share', 'mlruns', '0',
+                        # 'ae72265a0a17473f993f78ab239c2f2f', 'artifacts', 'model')
+                        source_path_parts = Path(model_version.source).parts
+                        # leave pure mlflow subpath, e.g.:
+                        # (0, ae72265a0a17473f993f78ab239c2f2f, artifacts, model)
+                        mlruns_idx = source_path_parts.index(model_version.run_id) - 1
+                        mlflow_source_parts = source_path_parts[mlruns_idx:]
+                        storage_art_uri = URL(
+                            f"{mlflow_storage_root}/{'/'.join(mlflow_source_parts)}"
+                        )
+                        registered_model = _DeployedModel(
+                            image=deploy_image_ref,
+                            model_storage_uri=storage_art_uri,
+                            model_name=model.name,
+                            model_stage=model_version.current_stage,
+                            model_version=model_version.version,
+                            source_run_id=model_version.run_id,
+                            deployment_namespace=seldon_deployment_ns,
+                        )
+                        deployed_model = seldon_models.get(registered_model.name)
+                        if not deployed_model or not deployed_model.is_same_version(
+                            registered_model
+                        ):
+                            registered_model.need_redeploy = True
+                        else:
+                            registered_model.need_redeploy = False
+                        mlflow_models[registered_model.name] = registered_model
+
+                # deploy models in Seldon
+                for model_name in mlflow_models.keys():
```

Comment:
minor: same here: `for k in dictionary: pass` is the same as `for k in dictionary.keys(): pass`